rangeTester.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <assert.h>
+#include "rangeFinder.h"
+
+void Test_Case0()
+{
+  int currentRange[4] = {10,1,4,6};
+  int sortedRange[4] = {1,10,4,6};
+  int length = sizeof(currentRange)/sizeof(currentRange[0]);
+  sortGivenIndexPosition(currentRange, length, 0);
+  
+  for(int index = 0; index < length; index++)
+  {
+    assert(currentRange[index] == sortedRange[index]);
+  }
+}
+
+void Test_Case1()
+{
+  int currentRange[4] = {10,1,4,6};
+  int sortedRange[4] = {10,1,4,6};
+  int length = sizeof(currentRange)/sizeof(currentRange[0]);
+  sortGivenIndexPosition(currentRange, length, 1);
+  
+  for(int index = 0; index < length; index++)
+  {
+    assert(currentRange[index] == sortedRange[index]);
+  }
+}
+
+void Test_Case2()
+{
+  int length, range;
+  
+  int currentRange1[4] = {10,1,4,6};
+  length = sizeof(currentRange1)/sizeof(currentRange1[0]);
+  
+  range = get_noofRanges(currentRange1, length);
+  assert(range == 4);
```

Comment:
Along with number of ranges , if the values can be verified, then the test case is effective. It also adds readability and gives an immediate glimpse of what is the expected output.
Otherwise, it might take time to understand what code does and then to verify if the test cases are right.

Reply:
Test cases updated.
Could you please check and provide comment